python_scripts/trees.py

```diff
@@ -469,16 +473,19 @@ def information_gain(labels_below_threshold, labels_above_threshold):
 
 # %% [markdown]
 # The implementation in scikit-learn gives similar results: 43.25 mm. The
-# slight difference are only due to some low-level implementation details.
+# slight difference is just due to some low-level implementation details.
 #
 # As we previously explained, the split mechanism will be repeated several
-# times (until we don't have any classification error on the training set). In
-# the above example, it corresponds to increasing the `max_depth` parameter.
+# times (until there is no classification error on the training set,
+# i.e., all final partitions consist of only one class). In
+# the above example, it corresponds to setting the `max_depth` parameter to
+# `None`. This allows the algorithm to keep making splits until the final
+# partitions are pure.
```

Comment:
Tricky but should we be talking about splitting until no error ... (as this would be un-advisable due to overfitting) ? Is this the 'default' action of the tree algorithm?

Reply:
For the moment I'm fine with that. It seems to me it is clearly explain here.

Reply:
It is clearly explained. My concern is that doing this is often not advisable as it leads to overfitting. 

Is it too verbose to say something like:
'until there is no classification error on the training set or one of the limiting parameters such as `max_depth` or `min_samples_leaf` has been reached.'  ?